tests/h/cli/commands/user_test.py

```diff
@@ -62,266 +58,190 @@ def test_it_adds_user_with_specific_authority(self, cli, cliconfig, signup_servi
+        result = invoke_cli(
+            user_cli.delete, ["--authority", "partner.org", user.username]
         )
 
         assert not result.exit_code
-        assert not db_session.query(models.User).filter_by(id=user.id).count()
+        user_delete_service.delete.assert_called_once_with(user)
 
     def test_it_errors_when_user_could_not_be_found(
-        self, cli, cliconfig, user, db_session
+        self, invoke_cli, user_delete_service
     ):
-        result = cli.invoke(user_cli.delete, [f"bogus_{user.username}"], obj=cliconfig)
+        result = invoke_cli(user_cli.delete, ["bogus_username"])
 
         assert result.exit_code == 1
-        assert db_session.query(models.User).filter_by(id=user.id).count() == 1
+        user_delete_service.delete.assert_not_called()
 
     def test_it_errors_when_user_with_specific_authority_could_not_be_found(
-        self, cli, cliconfig, user, db_session
+        self, invoke_cli, user, user_delete_service
     ):
-        result = cli.invoke(
-            user_cli.delete, ["--authority", "foo.com", user.username], obj=cliconfig
-        )
+        result = invoke_cli(user_cli.delete, ["--authority", "foo.com", user.username])
 
         assert result.exit_code == 1
-        assert db_session.query(models.User).filter_by(id=user.id).count() == 1
+        user_delete_service.delete.assert_not_called()
 
     @pytest.fixture
-    def user(self, db_session, factories):
-        user = factories.User()
-        db_session.flush()
-        return user
+    def user(self, factories):
+        return factories.User()
 
 
 @pytest.fixture
-def signup_service():
-    signup_service = mock.Mock(spec_set=["signup"])
-    return signup_service
-
-
-@pytest.fixture
-def hasher():
-    # Use a much faster hasher for testing purposes. DO NOT use as few as
-    # 5 rounds of bcrypt in production code under ANY CIRCUMSTANCES.
-    return CryptContext(
-        schemes=["bcrypt"],
-        bcrypt__ident="2b",
-        bcrypt__min_rounds=5,
-        bcrypt__max_rounds=5,
-    )
-
-
-@pytest.fixture
-def password_service(hasher):
-    password_service = UserPasswordService()
-    password_service.hasher = hasher
-    return password_service
-
-
-@pytest.fixture
-def user_delete_service(pyramid_request, annotation_delete_service):
-    return UserDeleteService(pyramid_request, annotation_delete_service)
-
-
-@pytest.fixture
-def annotation_delete_service(pyramid_config):  # pylint:disable=unused-argument
-    service = mock.create_autospec(
-        AnnotationDeleteService, spec_set=True, instance=True
-    )
-    return service
-
-
-@pytest.fixture
-def pyramid_config(
-    pyramid_config, signup_service, password_service, user_delete_service
-):
-    pyramid_config.register_service(signup_service, name="user_signup")
-    pyramid_config.register_service(password_service, name="user_password")
-    pyramid_config.register_service(user_delete_service, name="user_delete")
-    return pyramid_config
```

Comment:
None of this is required if we use the conventional fixtures in the normal way. You'll notice that we have a mixture of mocked and non-mocked services here which required weird work arounds like the hasher fixture above.

This fixture also coupled all of the tests unnecessarily. Each of these services is only relevant to one set of tests, but you wouldn't know it from looking at this lot.